### PR TITLE
extractioncomponentbase

### LINKED ISSUE #19: v1.10-Extract store object from Article fn. component to independent container component

### PR DESCRIPTION
**Pull request by Mhtshum**
fix #19 
- add type checking using props-types library for Article Library

### CODE REVIEW

File: lib/components/Article.js [hunk 1]
```diff
@@ -29,8 +29,8 @@ const styles = {
 const dateDisplay = (dateString) => new Date(dateString).toDateString();
 
 // {} these are used to access inner property of context object
-const Article = (props, { store }) => {
-	const { article } = props;
+const Article = (props) => {
+	const { article, store } = props;
 	const author = store.lookupAuthor(article.authorId);
 	return (
```
Comment (MhtshmZubair) [inline lib/components/Article.js:35]: Make sure testData is getting passed by console.log

**Review by MhtshmZubair — CHANGES_REQUESTED** [review]
See why test is failing use console.log to see

File: lib/components/Article.js (hunk 1 above)
Comment (Mhtshum) [reply lib/components/Article.js:35]: it was actually failing tests

File: lib/components/__tests__/ArticleList.js [hunk 2]
```diff
@@ -18,14 +18,14 @@ describe('ArticleList', () => {
...
 			<ArticleList
-			    { ...testProps }
+			    {...testProps}
 			/>
 		);
 		
-		//console.log(wrapper.node.props.children);
+		console.log(wrapper.node.props.children);
 		
-		expect(wrapper.find('Article').length).toBe(2);
```
Comment (Mhtshum) [inline lib/components/__tests__/ArticleList.js:27]: We need to find for ArticleContainer

File: lib/components/__tests__/ArticleList.js [hunk 3]
```diff
@@ -18,14 +18,14 @@ describe('ArticleList', () => {
 	it('render correctly', () => {
 		const wrapper = shallow(
 			<ArticleList
-			    { ...testProps }
+			    {...testProps}
 			/>
 		);
 		
-		//console.log(wrapper.node.props.children);
+		console.log(wrapper.node.props.children);
```
Comment (MhtshmZubair) [inline lib/components/__tests__/ArticleList.js]: Make sure to comment or remove the console statement still failing test

**Review by MhtshmZubair — CHANGES_REQUESTED** [review]

File: lib/components/__tests__/ArticleList.js (hunk 3 above)
Comment (Mhtshum) [reply lib/components/__tests__/ArticleList.js]: ohhhhhh it is mine bad

**Review by MhtshmZubair — CHANGES_REQUESTED** [review]
tests are failing because of old snapshot now

**Comment by Mhtshum** [comment]
> 
> 
> tests are failing because of old snapshot now

Now checks are passing so do the tests

**Review by MhtshmZubair — APPROVED** [review]
Changes are good!

**Pull request merged**